### ISSUE TITLE
Way to utilize `loss_weights` in custom `fit` method!

### ISSUE THREAD

**Issue by innat** [keras-team/tf-keras#356]
I'm going through this documents ([customizing_what_happens_in_fit](https://keras.io/guides/customizing_what_happens_in_fit/) by @fchollet), and eventually feel like getting stuck if I want to weighted the loss value. For example, observe the following code

```python
class CustomModel(keras.Model):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.loss_tracker = keras.metrics.Mean(name="loss")
        self.mae_metric = keras.metrics.MeanAbsoluteError(name="mae")

    def train_step(self, data):
        x, y = data

        with tf.GradientTape() as tape:
            y_pred = self(x, training=True)  # Forward pass
            # Compute our own loss
            loss = keras.losses.mean_squared_error(y, y_pred)

        # Compute gradients
        trainable_vars = self.trainable_variables
        gradients = tape.gradient(loss, trainable_vars)

        # Update weights
        self.optimizer.apply_gradients(zip(gradients, trainable_vars))

        # Compute our own metrics
        self.loss_tracker.update_state(loss)
        self.mae_metric.update_state(y, y_pred)
        return {"loss": self.loss_tracker.result(), "mae": self.mae_metric.result()}

    @property
    def metrics(self):
        # We list our `Metric` objects here so that `reset_states()` can be
        # called automatically at the start of each epoch
        # or at the start of `evaluate()`.
        # If you don't implement this property, you have to call
        # `reset_states()` yourself at the time of your choosing.
        return [self.loss_tracker, self.mae_metric]
```
```python
# Construct an instance of CustomModel
inputs = keras.Input(shape=(32,))
outputs = keras.layers.Dense(1)(inputs)
model = CustomModel(inputs, outputs)

# We don't passs a loss or metrics here.
model.compile(optimizer="adam")

# Just use `fit` as usual -- you can use callbacks, etc.
x = np.random.random((1000, 32))
y = np.random.random((1000, 1))
model.fit(x, y, epochs=5)
```

In the above, no loss is passed in `model.compile` method, because, we're manually computing loss value inside the `train_step` method. That's great. Now, let's assume I've more than one output from model and want to weighted them towards the final loss. In such case, what would be the approach to do so?

In `model.compile`, it has `loss_weights` arguments that can be used but for above cases, how to achieve this?

```python
Model.compile(
    optimizer="rmsprop",
    loss=None,
    metrics=None,
    loss_weights=None,
    weighted_metrics=None,
    run_eagerly=None,
    steps_per_execution=None,
    jit_compile=None,
    **kwargs
)
```

In the above mention [document](https://keras.io/guides/customizing_what_happens_in_fit/), it somewhat discuss about `class_weights` and `sample_weights`, but not about `loss_weights`.

**Comment by jgonsie** [comment]
@innat I reached the same doubt as you. Your proposed workaround works, but I think it is easier to access the loss_weights created when you .compile() your model through the variable self.compiled_loss._loss_weights
